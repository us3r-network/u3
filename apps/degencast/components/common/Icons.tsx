import { Info, LucideIcon, MoonStar, Sun } from 'lucide-react-native';
import { cssInterop } from 'nativewind';

function interopIcon(icon: LucideIcon) {
  cssInterop(icon, {
    className: {
      target: 'style',
      nativeStyleToProp: {
        color: true,
        opacity: true,
      },
    },
  });
}

interopIcon(Info);
interopIcon(MoonStar);
interopIcon(Sun);

export { Info, MoonStar, Sun };
