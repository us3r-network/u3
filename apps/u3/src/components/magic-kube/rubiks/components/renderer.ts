import { WebGLRenderer } from 'three';

const createRenderer = () => {
  const renderer = new WebGLRenderer({ antialias: true });
  renderer.setClearColor(0x000000, 0);

  return renderer;
};

export default createRenderer;
