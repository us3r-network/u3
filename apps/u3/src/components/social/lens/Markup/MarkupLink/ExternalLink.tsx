export interface MarkupLinkProps {
  href?: string;
  title?: string;
}

export default function ExternalLink({ href, title = href }: MarkupLinkProps) {
  if (!href) {
    return null;
  }

  if (!href.includes('://')) {
    href = `https://${href}`;
  }

  return (
    <a
      href={href}
      onClick={(e) => e.stopPropagation()}
      target={href.includes(window.location.host) ? '_self' : '_blank'}
      rel="noopener noreferrer"
    >
      {title}
    </a>
  );
}
