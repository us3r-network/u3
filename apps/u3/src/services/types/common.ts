/*
 * @Author: shixuewen friendlysxw@163.com
 * @Date: 2022-12-01 14:05:16
 * @LastEditors: shixuewen friendlysxw@163.com
 * @LastEditTime: 2023-03-06 18:40:30
 * @Description: file description
 */
import { ApiResp } from '.';

export enum ChainType {
  EVM = 'EVM',
  SOLANA = 'SOLANA',
  BSC = 'BSC',
  MATIC = 'MATIC',
  APTOS = 'APTOS',
}

export enum OrderBy {
  EARLIEST = 'EARLIEST',
  TRENDING = 'TRENDING',
  NEWEST = 'NEWEST',
  FORU = 'FORU',
}

export enum Platform {
  GALXE = 'GALXE',
  NOOX = 'NOOX',
  POAP = 'POAP',
  QUEST3 = 'QUEST3',
  RABBIT_HOLE = 'RABBIT_HOLE',
  LINK3 = 'LINK3',
}

export enum Reward {
  BADGE = 'BADGE',
  NFT = 'NFT',
  TOKEN = 'TOKEN',
  WL = 'WL',
}

export enum TagType {
  SOCIAL = 'social',
  TRANSACTION = 'transaction',
  EXCHANGE = 'exchange',
  COLLECTIBLE = 'collectible',
  DONATION = 'donation',
  GOVERNANCE = 'governance',
}

export enum PlatformLogo {
  GALXE = 'https://galxe.com/favicon.ico',
  NOOX = 'https://noox.world/favicon.ico',
  POAP = 'https://poap.xyz/favicon-32x32.dba5403f.png',
  QUEST3 = 'https://quest3.xyz/favicon.ico',
}

export enum PlatformType {
  EVENT = 'EVENT',
  CONTENT = 'CONTENT',
}
export type PlatformsItemResponse = {
  platform: string;
  platformLogo: string;
  platformUrl: string;
  number: number;
  type: PlatformType;
};
export type PlatformsResponse = ApiResp<Array<PlatformsItemResponse>>;

export type ConfigTopicsChain = {
  chainEnum: string;
  image: string;
  link: string;
  name: string;
};
export type ConfigTopicsType = {
  name: string;
  num: number;
  image?: string;
};
export type ConfigTopics = {
  eventRewards: ConfigTopicsType[];
  eventTypes: ConfigTopicsType[];
  projectTypes: string[];
  dappTypes: ConfigTopicsType[];
  contentTypes: string[];
  langs: string[];
  chains: ConfigTopicsChain[];
  contentTags: ConfigTopicsType[];
};
export type ConfigTopicsResponse = ApiResp<ConfigTopics>;
