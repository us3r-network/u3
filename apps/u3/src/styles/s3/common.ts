import { css } from 'styled-components';

const Modal = css`
  .react-aria-ModalOverlay {
    position: fixed;
    top: 0;
    left: 0;
    width: 100vw;
    height: 100vh;
    background-color: rgba(0, 0, 0, 0.5);
    display: flex;
    align-items: center;
    justify-content: center;

    &[data-entering] {
      animation: fade 200ms;
    }

    &[data-exiting] {
      animation: fade 150ms reverse ease-in;
    }

    .react-aria-Modal {
      min-width: 380px;
      border-radius: 20px;
      background: rgb(27, 30, 35);
      padding: 20px;
      box-shadow: 0 8px 20px rgba(0 0 0 / 0.1);
      outline: none;

      &[data-entering] {
        animation: zoom 300ms cubic-bezier(0.175, 0.885, 0.32, 1.275);
      }

      .react-aria-Dialog {
        outline: none;

        .react-aria-Heading {
          line-height: 1em;
          margin-top: 0;
          font-style: italic;
          font-weight: 700;
          font-size: 24px;
          line-height: 28px;
          color: rgb(255, 255, 255);
        }
      }
    }
  }

  @keyframes fade {
    from {
      opacity: 0;
    }

    to {
      opacity: 1;
    }
  }

  @keyframes zoom {
    from {
      transform: scale(0.8);
    }

    to {
      transform: scale(1);
    }
  }
`;

const UseAvatar = css`
  [data-us3r-component='UserAvatar'] {
    display: inline-block;
    width: 32px;
    height: 32px;
    & > img {
      width: 100%;
      height: 100%;
      border-radius: 50%;
    }
  }
`;

const LoadingSpokes = css`
  [data-common-element='LoadingSpokes'] {
    fill: #718096;
  }
`;

export default css`
  ${Modal}
  ${UseAvatar}
  ${LoadingSpokes}

  .react-aria-Button,.react-aria-Link {
    outline: none;

    &[data-hovered] {
      outline: none;
    }

    &[data-pressed] {
      outline: none;
    }

    &[data-focus-visible] {
      outline: none;
    }
  }
`;
