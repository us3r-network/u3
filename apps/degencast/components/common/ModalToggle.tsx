import { router } from 'expo-router';
import { Pressable, View } from 'react-native';
import { CircleUserRound } from './Icons';
import { cn } from '@/lib/utils';

export function ModalToggle() {
  return (
    <Pressable
      onPress={() => {
        router.push('/modal');
      }}
      className='web:ring-offset-background web:transition-colors web:focus-visible:outline-none web:focus-visible:ring-2 web:focus-visible:ring-ring web:focus-visible:ring-offset-2'
    >
      {({ pressed }) => (
        <View
          className={cn(
            'flex-1 aspect-square justify-center items-end pt-0.5 web:pl-4',
            pressed && 'opacity-70'
          )}
        >
          <CircleUserRound className='text-foreground' size={24} strokeWidth={1.25} />
        </View>
      )}
    </Pressable>
  );
}
