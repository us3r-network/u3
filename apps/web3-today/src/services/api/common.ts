/*
 * @Author: shixuewen friendlysxw@163.com
 * @Date: 2022-12-14 13:03:00
 * @LastEditors: shixuewen friendlysxw@163.com
 * @LastEditTime: 2023-01-09 18:19:09
 * @Description: file description
 */
// 获取所有Platform
import { ConfigTopicsResponse, PlatformsResponse } from '../types/common';
import request, { RequestPromise } from './request';

export function getAllPlatform(): RequestPromise<PlatformsResponse> {
  return request({
    url: `/uniprojects/platforms`,
    method: 'get',
  });
}
export function getConfigsTopics(): RequestPromise<ConfigTopicsResponse> {
  return request({
    url: `/configs/topics`,
    method: 'get',
  });
}
