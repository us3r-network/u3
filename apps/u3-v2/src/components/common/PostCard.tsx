import styled, { StyledComponentPropsWithRef } from 'styled-components'
import { SocailPlatform } from '../../api'
import { useMemo } from 'react'
import LensIcon from './icons/LensIcon'
import FarcasterIcon from './icons/FarcasterIcon'
import dayjs from 'dayjs'
import PostLike from './PostLike'
import PostReply from './PostReply'
import PostReport from './PostReport'

export type PostCardData = {
  platform: SocailPlatform
  avatar: string
  name: string
  handle: string
  createdAt: string | number
  content?: string
  totalLikes?: number
  totalReplies?: number
  totalReposts?: number
  likeAvatars?: string[]
}
interface PostCardProps {
  data: PostCardData
  contentRender?: () => JSX.Element
  liked?: boolean
  replied?: boolean
  reposted?: boolean
  liking?: boolean
  replying?: boolean
  reposting?: boolean
  likeAction?: () => void
  replyAction?: () => void
  repostAction?: () => void
  showActions?: boolean
}
export default function PostCard({
  data,
  contentRender,
  liked,
  replied,
  reposted,
  liking,
  replying,
  reposting,
  likeAction,
  replyAction,
  repostAction,
  showActions = true,
  ...wrapperProps
}: StyledComponentPropsWithRef<'div'> & PostCardProps) {
  return (
    <PostCardWrapper {...wrapperProps}>
      <PostCardUserInfo data={data} />
      <PostCardContentWrapper>
        {contentRender ? contentRender() : data?.content}
      </PostCardContentWrapper>
      {showActions && (
        <PostCardActionsWrapper>
          <PostLike
            totalLikes={data?.totalLikes || 0}
            likeAvatars={[]}
            liking={liking}
            liked={liked}
            likeAction={likeAction}
          />
          <PostReply
            totalReplies={data?.totalReplies || 0}
            replying={replying}
            replied={replied}
            replyAction={replyAction}
          />
          <PostReport
            totalReposts={data?.totalReposts || 0}
            reposting={reposting}
            reposted={reposted}
            repostAction={repostAction}
          />
        </PostCardActionsWrapper>
      )}
    </PostCardWrapper>
  )
}

export const PostCardWrapper = styled.div`
  background: #212228;
  padding: 20px;
  box-sizing: border-box;
  display: flex;
  flex-direction: column;
  gap: 20px;
`

export type PostCardUserInfoData = {
  platform: SocailPlatform
  avatar: string
  name: string
  handle: string
  createdAt: string | number
}
export function PostCardUserInfo({
  data,
  ...wrapperProps
}: StyledComponentPropsWithRef<'div'> & { data: PostCardUserInfoData }) {
  const PlatFormIcon = useMemo(() => {
    switch (data.platform) {
      case SocailPlatform.Lens:
        return <LensIcon />
      case SocailPlatform.Farcaster:
        return <FarcasterIcon />
      default:
        return null
    }
  }, [data.platform])
  return (
    <PostCardUserInfoWrapper {...wrapperProps}>
      <Avatar src={data.avatar} />
      <PostCardUserInfoCenter>
        <Name>
          {data.name} {PlatFormIcon}
        </Name>
        <Handle>
          @{data.handle} . {dayjs(data.createdAt).fromNow()}
        </Handle>
      </PostCardUserInfoCenter>
    </PostCardUserInfoWrapper>
  )
}

const PostCardUserInfoWrapper = styled.div`
  display: flex;
  align-items: flex-start;
  gap: 10px;
`
const PostCardUserInfoCenter = styled.div`
  width: 0;
  flex: 1;
  display: flex;
  flex-direction: column;
  gap: 5px;
`
const Avatar = styled.img`
  width: 40px;
  height: 40px;
  flex-shrink: 0;
  border-radius: 50%;
  object-fit: cover;
`
const Name = styled.div`
  display: flex;
  align-items: center;
  gap: 10px;

  color: #fff;
  font-family: Baloo Bhai 2;
  font-size: 16px;
  font-style: normal;
  font-weight: 700;
  line-height: normal;
`
const Handle = styled.div`
  display: flex;
  align-items: center;
  gap: 5px;

  color: #9c9c9c;
  font-family: Baloo Bhai 2;
  font-size: 12px;
  font-style: normal;
  font-weight: 400;
  line-height: normal;
`

export const PostCardContentWrapper = styled.div`
  color: #fff;
  font-family: Baloo Bhai 2;
  font-size: 16px;
  font-style: normal;
  font-weight: 400;
  line-height: 25px; /* 156.25% */
`
export const PostCardActionsWrapper = styled.div`
  display: flex;
  align-items: center;
  gap: 15px;
`

export const PostCardImgWrapper = styled.div`
  display: flex;
  gap: 10px;

  > img {
    max-width: 30%;
  }
`
