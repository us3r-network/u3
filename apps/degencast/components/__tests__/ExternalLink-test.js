import * as React from 'react';
import renderer from 'react-test-renderer';

import { ExternalLink } from '../common/ExternalLink';

it(`renders correctly`, () => {
  const tree = renderer.create(<ExternalLink href="http://u3.xyz">U3</ExternalLink>).toJSON();

  expect(tree).toMatchSnapshot();
});
