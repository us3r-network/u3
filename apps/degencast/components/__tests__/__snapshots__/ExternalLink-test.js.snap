// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Text
  href="http://u3.xyz"
  onPress={[Function]}
  role="link"
>
  U3
</Text>
`;
